frontend/src/lib/WorkflowParser.ts

```diff
@@ -163,10 +163,17 @@ export default class WorkflowParser {
   // Returns a list of output paths for the entire workflow, by searching all nodes in
   // the workflow, and parsing outputs for each.
   public static loadAllOutputPaths(workflow: Workflow): StoragePath[] {
-    let outputPaths: StoragePath[] = [];
+    return this.loadAllOutputPathsWithStepNames(workflow).map(entry => entry[1]);
+  }
+
+  // Returns a list of object mapping a step name to output path for the entire workflow,
+  // by searching all nodes in the workflow, and parsing outputs for each.
+  public static loadAllOutputPathsWithStepNames(workflow: Workflow): Array<[string, StoragePath]> {
```

Comment:
I have a slight preference for objects over tuples for this kind of thing.
For example, without coming to this function, I wouldn't know what the first string is, and in the above function `loadAllOutputPaths`, it's not immediately clear why it's safe to assume `entry[1]` is indeed a `StoragePath`.

Still, you can leave it if you prefer it

Reply:
No I prefer objects too. Done.